service/history/wrappers/ratelimited/handler_generated.go

```diff
@@ -0,0 +1,314 @@
+	return h.wrapped.RespondActivityTaskCompleted(ctx, hp1)
+}
+
+func (h *historyHandler) RespondActivityTaskFailed(ctx context.Context, hp1 *types.HistoryRespondActivityTaskFailedRequest) (err error) {
+	return h.wrapped.RespondActivityTaskFailed(ctx, hp1)
+}
+
+func (h *historyHandler) RespondCrossClusterTasksCompleted(ctx context.Context, rp1 *types.RespondCrossClusterTasksCompletedRequest) (rp2 *types.RespondCrossClusterTasksCompletedResponse, err error) {
+	return h.wrapped.RespondCrossClusterTasksCompleted(ctx, rp1)
+}
+
+func (h *historyHandler) RespondDecisionTaskCompleted(ctx context.Context, hp1 *types.HistoryRespondDecisionTaskCompletedRequest) (hp2 *types.HistoryRespondDecisionTaskCompletedResponse, err error) {
+	return h.wrapped.RespondDecisionTaskCompleted(ctx, hp1)
+}
+
+func (h *historyHandler) RespondDecisionTaskFailed(ctx context.Context, hp1 *types.HistoryRespondDecisionTaskFailedRequest) (err error) {
+	return h.wrapped.RespondDecisionTaskFailed(ctx, hp1)
+}
+
+func (h *historyHandler) ScheduleDecisionTask(ctx context.Context, sp1 *types.ScheduleDecisionTaskRequest) (err error) {
+	return h.wrapped.ScheduleDecisionTask(ctx, sp1)
+}
+
+func (h *historyHandler) SignalWithStartWorkflowExecution(ctx context.Context, hp1 *types.HistorySignalWithStartWorkflowExecutionRequest) (sp1 *types.StartWorkflowExecutionResponse, err error) {
+
+	if hp1 == nil {
+		err = validate.ErrRequestNotSet
+		return
+	}
+
+	if hp1.GetDomainUUID() == "" {
+		err = validate.ErrDomainNotSet
+		return
+	}
+
+	if hp1.SignalWithStartRequest.GetWorkflowID() == "" {
+		err = validate.ErrWorkflowIDNotSet
+		return
+	}
+
+	h.workflowIDCache.AllowExternal(hp1.SignalWithStartRequest.GetWorkflowID(), hp1.GetDomainUUID())
+	return h.wrapped.SignalWithStartWorkflowExecution(ctx, hp1)
+}
+
+func (h *historyHandler) SignalWorkflowExecution(ctx context.Context, hp1 *types.HistorySignalWorkflowExecutionRequest) (err error) {
+
+	if hp1 == nil {
+		err = validate.ErrRequestNotSet
+		return
+	}
+
+	if hp1.GetDomainUUID() == "" {
+		err = validate.ErrDomainNotSet
+		return
+	}
+
+	if hp1.SignalRequest.GetWorkflowExecution().GetWorkflowID() == "" {
+		err = validate.ErrWorkflowIDNotSet
+		return
+	}
+
+	h.workflowIDCache.AllowExternal(hp1.SignalRequest.GetWorkflowExecution().GetWorkflowID(), hp1.GetDomainUUID())
+	return h.wrapped.SignalWorkflowExecution(ctx, hp1)
+}
+
+func (h *historyHandler) Start() {
+	h.wrapped.Start()
+	return
+}
+
+func (h *historyHandler) StartWorkflowExecution(ctx context.Context, hp1 *types.HistoryStartWorkflowExecutionRequest) (sp1 *types.StartWorkflowExecutionResponse, err error) {
+
+	if hp1 == nil {
+		err = validate.ErrRequestNotSet
+		return
+	}
+
+	if hp1.GetDomainUUID() == "" {
+		err = validate.ErrDomainNotSet
+		return
+	}
+
+	if hp1.StartRequest.GetWorkflowID() == "" {
+		err = validate.ErrWorkflowIDNotSet
+		return
+	}
+
+	h.workflowIDCache.AllowExternal(hp1.StartRequest.GetWorkflowID(), hp1.GetDomainUUID())
```

Comment:
I am guessing you're not actually acting on the result of the rate-limit but just emitting a side-effect for visibility, is that correct?